### ISSUE TITLE
Overview article is missing key workflow concepts

### ISSUE THREAD

**Issue by ianw1223** [dotnet/EntityFramework.Docs#4379]
This overview article is missing some key concepts that are essential to reasoning about and understanding migrations.  Here is roughly what I believe is missing. The goal is to provide just enough of a framework so the reader can see where the subsequent details fit into the overall workflow:

**Migrations Workflow Key Concepts:**
Once the model has been changed in your code, you will run command line EF Core Tools to generate the migration. This only produces C# code in your project which should be under source control with your project. It does not yet change your database. Next, the EF Core Tools commands can be used to actually update the database. It will (behind the scenes) use the migration to generate the SQL scripts needed to make the schema changes, and then run those scripts. That is appropriate for development situations and very simple production scenarios. In other cases you may want to just generate and save those migration scripts as SQL files. There are EF Core Tools commands to do that as well.

To keep track of the history of migrations applied, EF Core Tools will create and use a history table in your database. The combination of the current database schema, the history table, and the in-project C# migration files is important to keeping your migrations working smoothly. Typically none of those should be changed manually. You will update you model, and use the EF Core tools to propagate those changes to the schema, history, and migrations.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6e28b0a8-b7e4-4c07-9e60-0df403bbee28
* Version Independent ID: 12300b31-6f7b-b75c-970d-693365e77013
* Content: [Migrations Overview - EF Core](https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/?tabs=dotnet-core-cli)
* Content Source: [entity-framework/core/managing-schemas/migrations/index.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/managing-schemas/migrations/index.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @bricelam
* Microsoft Alias: **avickers**

**Comment by roji** [comment]
@ianw1223 one guiding principle when writing this overview page was to not to try to cover all concepts in a big, up-front theoretical wall-of-text, but rather to get to the "getting started" walkthrough as quickly as possible; we know that dense, conceptual up-front introductions make it harder for new users to learn a new thing, and frequently leads to them abandoning it.

I generally think that it's better for the concepts to come out of a practical, hands-on tutorial rather than be defined/explained up-front. For example, the "Create the first migration" section shows `dotnet ef migrations add`, explains that it generates some files, and encourages the user to examine those files. This should make it clear that the database isn't yet updated at that point, but rather in the next step ("create your database and schema").

**Comment by ianw1223** [comment]
TL;DR: Revised suggestions at the end.

@roji I can appreciate the desire to avoid lots of up-front text, but having just struggled through the process of getting my head around migrations and solving a problem, I can say that deducing this overall workflow was not easy. There are a lot of new details within each section, and what I needed up front was a conceptual skeleton to pin these things things to, rather than relying on what at the time is behind-the-scenes magic. In particular, the big picture about what to expect in terms of C# vs SQL and how it relates to the workflow was critical.

It is also fascinating to re-read the overview material that you specifically referenced -- now from a perspective of understanding, and to consider why it did not click with me when I did not understand the workflow.

At the risk of going on about this, let me share the noob experience on that overview page. This will seem obtuse and argumentative, but I'm trying to illustrate the gaps that I struggled with in the hope that it ends up being constructive:

In the "Create your first migration" section we invoke a command that puts .cs files in our project. Okay. When/how will I use these when my project runs? Will they get invoked somehow by the NuGet packages at runtime?  Not sure. Just park that for now. Next section: "Create your database and schema". We are running another command that changes/creates the database. Okay. Is the .cs migration code in my project connected to that, or something separately needed at run time? Hmm ... no SQL was involved, either! (But EF did secretly write SQL for you, and run it.)  At this point, what is going on is still not clear.

Let's turn step-by-step magic into the ability to reason about it: 

How about adding a sentence or two in each section, noting that identifying the file types adds meaning for those who are using the overview as reference rather than a follow-along tutorial.

Create your first migration
....skip that for now. **Those C# files define the migration and will only be used by EF Core to generate SQL that will modify the database**.

Create your database and schema
**Check your database and note the new tables.**  ... and you didn't need to write a single line of SQL.  **EF Core used the C# files that define the migration to produce and run SQL scripts to create your database.** Note that ..... less suitable for production environments **where you will want to generate the SQL and separately manage its use** - see the Applying Migrations ...

Hitting us over the head with these basic details would be very helpful.

I know that documentation design is a challenge. Thanks for considering my perspective.

**Comment by roji** [comment]
@ianw1223 thanks for these suggestions - they don't at all come across as obtuse/argumentative, at least some of these make sense to me. I'll look at integrating some into the page.
